modules/copy-template/pkg/templates/template-provider_test.go

```diff
@@ -80,6 +81,52 @@ var _ = Describe("Template provider", func() {
 					Expect(updatedTemplate.Labels[key]).To(Equal(""))
 				}
 			}
+
+		})
+
+		It("should remove Common template informations from VM", func() {
+			tProvider := &templates.TemplateCreator{}
+
+			vm := &kubevirtv1.VirtualMachine{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						templates.VMFlavorAnnotation:   "test",
+						templates.VMOSAnnotation:       "test",
+						templates.VMWorkloadAnnotation: "test",
+						templates.VMDomainLabel:        "test",
+						templates.VMSizeLabel:          "test",
+						"someOtherLabel":               "test",
+					},
+					Annotations: map[string]string{
+						templates.VMFlavorAnnotation:   "test",
+						templates.VMOSAnnotation:       "test",
+						templates.VMWorkloadAnnotation: "test",
+						templates.VMDomainLabel:        "test",
+						templates.VMSizeLabel:          "test",
+						"someOtherLabel":               "test",
+					},
+				},
+			}
+
+			tProvider.UpdateVMMetaObject(vm)
+			Expect(len(vm.GetLabels())).To(Equal(1))
+			Expect(len(vm.GetAnnotations())).To(Equal(1))
+
+			for key, val := range vm.GetLabels() {
+				if key == "someOtherLabel" {
+					Expect(vm.Labels[key]).To(Equal(val))
+				} else {
+					Expect(vm.Labels[key]).To(Equal(""))
```

Comment:
Here, you could use `Expect(vm.Labels).ToNot(HaveKey(key))`, which would not pass if the label exists, but is `""`.